lib/safira/roulette/roulette.ex

```diff
@@ -196,4 +205,133 @@ defmodule Safira.Roulette do
   def change_attendee_prize(%AttendeePrize{} = attendee_prize) do
     AttendeePrize.changeset(attendee_prize, %{})
   end
+
+  @doc """
+  Transaction that take a number of tokens from an attendee,
+  apply a probability-based function for "spinning the wheel",
+  and give the price to the attendee.
+  """
+  def spin(attendee) do
+    Multi.new()
+    # remove tokens from attendee to spin the wheel
+    |> Multi.update(
+      :attendee,
+      Attendee.update_token_balance_changeset(attendee, %{
+        token_balance: attendee.token_balance - Application.fetch_env!(:safira, :roulette_cost)
+      })
+    )
+    # prize after spinning
+    |> Multi.run(:prize, fn _repo, _changes -> {:ok, spin_prize()} end)
+    # get an attendee_prize if exists one, or build one
+    |> Multi.run(:attendee_prize, fn _repo, %{attendee: attendee, prize: prize} ->
+      {:ok,
+       get_keys_attendee_prize(attendee.id, prize.id) ||
+         %AttendeePrize{attendee_id: attendee.id, prize_id: prize.id, quantity: 0}}
+    end)
```

Comment:
Just checking, the `attendee` used at line 226 is the one the `spin` function received by argument  (thus being not the updated version) and not the updated one which resulted from the Multi.Update operation, right?

Reply:
I know this doens't make any difference regarding your code, I was just curious to see if there was any way that elixir/Multi could detect that the attendee had already been updated in a previous operation.